### PR TITLE
New statusd_inetaddr2 statusbar template

### PR DESCRIPTION
**Pull request by knixeur**
It displays selected interfaces's IPv4 and IPv6 address in the
statusbar.

### CODE REVIEW

**Comment by gbonvehi** [comment]
I need to review how to parse the IP address as different ifconfig versions have different output.

**Comment by jeffpc** [comment]
It gets worse... ifconfig output can be *very* different on other UNIX systems.  E.g., from OpenIndiana (an illumos distro):

```
$ ifconfig 
lo0: flags=2001000849<UP,LOOPBACK,RUNNING,MULTICAST,IPv4,VIRTUAL> mtu 8232 index 1
	inet 127.0.0.1 netmask ff000000 
dhcp0: flags=1004843<UP,BROADCAST,RUNNING,MULTICAST,DHCP,IPv4> mtu 1500 index 2
	inet 10.10.1.34 netmask fffffe00 broadcast 10.10.1.255
```

OI has no `ip` command either.

**Comment by raboof** [comment]
Thanks for your contribution!

I don't have a big problem with accepting a 'contrib' script that does not work on all platforms, but perhaps we should specify which platforms are known to work at the top of the script?

Given that tool output may change even within a platform, it might be good to add a testcase to make sure future changes to this script won't break the output as it appears on your system. On the other hand we don't really have tests for the contrib scripts yet, so this might be a bit of extra work - it's up to you whether you'd like to take this on. mod_xinerama and mod_xrandr do have some tests you could use as inspiration.

**Comment by wilhelmy** [comment]
Just a note, I think some distros have ip(8) in /usr/sbin.

**Comment by raboof** [comment]
I see the latest commit here was 'WIP', can you ping when this is ready to go?

**Comment by knixeur** [comment]
Sure, I forgot about this PR but still use it. I'll try to make it more generic by detecting which tool is available.

File: contrib/statusd/statusd_inetaddr2.lua [hunk 1]
```diff
@@ -38,21 +38,23 @@ local inetaddr2_timer = statusd.create_timer()
...
     end
     while true do
         local line = f:read("*l")
         if line == nil then break end
         if (string.match(line, inet4_match)) then
-            inet4_addr = line:match("inet addr:(%d+%.%d+%.%d+%.%d+)")
-        elseif (string.match(line, inet4_match)) then
-            inet6_addr = line:match(("([a-fA-F0-9]*):"):rep(8):gsub(":$","$"))
+            inet4_addr = line:match("inet (%d+%.%d+%.%d+%.%d+)")
+            if inet4_addr == nil then inet4_addr = "none" end
```
Comment (wilhelmy) [inline contrib/statusd/statusd_inetaddr2.lua]: I think this can shrink to `inet4_addr = line:match(...) or "none"`

**Comment by knixeur** [comment]
I think this is ready now, you can choose to use ip addr or ifconfig and it will search for the binary in /sbin/ /usr/sbin/ /bin/, etc.
I could also use PATH environment value, but I think it's usable at it is.

**Comment by raboof** [comment]
Nice!

**Pull request merged**